src/puppetlabs/trapperkeeper/logging.clj

```diff
@@ -72,25 +72,26 @@
             (.toInt level))
        (.setLevel root level)))))
 
-(defn configure-logger-via-file!
-  "Reconfigures the current logger based on the supplied configuration
-  file."
-  [logging-conf-file]
-  {:pre [(string? logging-conf-file)]}
+(defn configure-logger!
+  "Reconfigures the current logger based on the supplied configuration."
+  [logging-conf]
+  {:pre [(#{String java.io.File java.net.URL java.io.InputStream org.xml.sax.InputSource} (type logging-conf))]}
```

Comment:
Another way of doing this would be

```clojure
  {:pre [(some #(instance? % logging-conf) (list String java.io.File java.net.URL ...))]}
```
This has typically been our preferred method of checking multiple types in a pre- or post-condition, since rather than checking that the type of the argument is matches exactly what is listed, it will instead check that the argument is an instance of the classes listed.

For example:

```clojure
(#{java.io.InputStream} (type (clojure.java.io/input-stream "path/to/logging-conf")))
=> nil
```
since `clojure.java.io/input-stream` creates a `java.io.BufferedInputStream` by default, which isn't explicitly listed in the set. However,

```clojure
(instance? java.io.InputStream (clojure.java.io/input-stream "path/to/logging-conf"))
=> true
```

Writing the precondition using `type` means that a `logging-conf` of type `BufferedInputStream` would cause this to raise an error, even though `doConfigure` will accept a `BufferedInputStream` without error.